conda_forge_tick/new_update_versions.py

```diff
@@ -0,0 +1,148 @@
+            else:
+                logger.debug(f"Upstream: Could not find version on {source.name}")
+                meta["bad"] = f"Upstream: Could not find version on {source.name}"
+        if not meta_yaml.get("bad"):
+            logger.debug("Upstream: unknown source")
+            meta["bad"] = "Upstream: unknown source"
+        return False
+
+
+# It's expected that your environment provide this info.
+CONDA_FORGE_TICK_DEBUG = os.environ.get("CONDA_FORGE_TICK_DEBUG", False)
+
+
+def new_update_upstream_versions(
+    gx: nx.DiGraph, sources: Iterable[AbstractSource] = None
+) -> None:
+    sources = (
+        (PyPI(), CRAN(), NPM(), ROSDistro(), RawURL(), Github())
+        if sources is None
+        else sources
+    )
+
+    _all_nodes = [t for t in gx.nodes.items()]
+    random.shuffle(_all_nodes)
+
+    # Inspection the graph object and node update:
+    # print(f"Number of nodes: {len(gx.nodes)}")
+    Node_count = 0
+
+    to_update = []
+    for node, node_attrs in tqdm.tqdm(_all_nodes):
+        attrs = node_attrs["payload"]
+        if attrs.get("bad") or attrs.get("archived"):
+            attrs["new_version"] = False
+            continue
+        to_update.append((node, attrs))
+
+    up_to = {}
+    for node, node_attrs in to_update:
+        # checking each node
+        with node_attrs as attrs:
+            up_to = {}
+
+            # verify the actual situation of the package;
+            actual_ver = str(attrs.get("version"))
+
+            # rude exception
+            if node == "ca-policy-lcg":
+                up_to["ca-policy-lcg"] = {
+                    "bad": attrs.get("bad"),
+                    "new_version": False,
+                    "new_version_attempts": attrs.get("new_version_attempts"),
+                    "new_version_errors": attrs.get("new_version_errors")
+                }
+                Node_count += 1
+                continue
+
+            # New verison request
+            try:
+                new_version = get_latest_version(node, attrs, sources)
+                attrs["new_version"] = new_version or attrs["new_version"]
+            except Exception as e:
+                try:
+                    se = repr(e)
+                except Exception as ee:
+                    se = "Bad exception string: {}".format(ee)
+                logger.warning(
+                    f"Warning: Error getting upstream version of {node}: {se}"
+                )
+                attrs["bad"] = "Upstream: Error getting upstream version"
+            else:
+                logger.info(
+                    f"# {Node_count:<5} - {node} - {attrs.get('version')} - {attrs.get('new_version')}",
+                )
+            up_to[f"{node}"] = {
+                "bad": attrs.get("bad"),
+                "new_version": attrs.get("new_version"),
+                "new_version_attempts": attrs.get("new_version_attempts"),
+                "new_version_errors": attrs.get("new_version_errors")
```

Comment:
The `new_version_attempts` and `new_version_errors` have to stay in the graph data.